athena-slack-member-analytics/src/main/java/com/amazonaws/connectors/athena/slack/util/SlackHttpUtility.java

```diff
@@ -0,0 +1,188 @@
+/*-
+ * #%L
+ * athena-slackapi-example
+ * %%
+ * Copyright (C) 2019 - 2020 Amazon Web Services
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.slack.util;
+
+import org.apache.http.*;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.message.BasicNameValuePair;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.util.EntityUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.InputStreamReader;
+import java.io.IOException;
+import java.io.BufferedReader;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.zip.GZIPInputStream;
+
+public class SlackHttpUtility {
+
+    private static final Logger logger = LoggerFactory.getLogger(SlackHttpUtility.class);
+
+    /**
+     * Makes an HTTP request using GET method to the specified URL.
+     *
+     * @param requestURL
+     *            the URL of the remote server
+     * @param headers HashMap<String,String> of request Headers
+     * @return An CloseableHttpResponse object
+     * @throws IOException
+     *             thrown if any I/O error occurred
+     */
+    public static CloseableHttpResponse doGetRequest(URIBuilder requestURI, HashMap<String, String> headers)
+            throws IOException, Exception {
+
+        logger.info("doGetRequest: enter - {}", requestURI.toString());
+        HttpGet httpGet = new HttpGet(requestURI.build());
+        for (String i : headers.keySet()) {
+            httpGet.setHeader(i,headers.get(i));
+        }
+
+        CloseableHttpClient client = HttpClients.createDefault();
+        CloseableHttpResponse response = client.execute(httpGet);
+        if (!isRequestOk(response)){
+            client.close();
+            throw new IOException("doGetRequest: response " + response.getStatusLine().getReasonPhrase());
```

Comment:
Why IOException? Seems like RuntimeException would be more appropriate

Reply:
changed to runtimeexception